python/adelphi/adelphi/nb.py

```diff
@@ -0,0 +1,132 @@
+
+class NbExporter(BaseExporter):
+
+    def __init__(self, cluster, props):
+
+        # Always disable anonymization when generating nosqlbench configs
+        self.props = props.copy()
+        self.props["anonymize"] = False
+
+        self.metadata = self.get_common_metadata(cluster, self.props)
+
+        all_keyspaces = self.get_keyspaces(cluster, self.props)
+        if len(all_keyspaces) > 1:
+            raise KeyspaceSelectionException("nosqlbench export doesn't support multiple keyspaces")
+        self.keyspace = next(iter(all_keyspaces.keys()))
+
+        if len(self.keyspace.tables) == 0:
+            raise TableSelectionException("Keyspace {} contains no tables".format(self.keyspace.name))
+        self.table = next(iter(self.keyspace.tables.values()))
+
+        log.info("Creating nosqlbench config for {}.{}".format(self.keyspace.name, self.table.name))
+
+
+    def export_all(self):
+        return self.export_schema()
+
+
+    def export_metadata(self):
+        return {k : self.metadata[k] for k in self.metadata.keys() if self.metadata[k]}
+
+
+    def export_schema(self):
+        return self.__build_schema()
+
+
+    # Requires custom impl because we don't want to support the notion of an arbitrary keyspace_id.
+    # We don't anonymize nosqlbench configs at all so just explicitly use keyspace name here.
+    def each_keyspace(self, ks_fn):
+        ks_fn(self.keyspace, self.keyspace.name)
+
+
+    def __build_rampup_statement(self):
+        return {"tags":{"name":"rampup-insert"}, "rampup-insert": build_insert_statements(self.keyspace, self.table)}
+
+
+    def __build_main_read_statement(self):
+        return {"tags":{"name":"main-select"}, "main-select": build_select_statements(self.keyspace, self.table)}
+
+
+    def __build_main_write_statement(self):
+        return {"tags":{"name":"main-insert"}, "main-insert": build_insert_statements(self.keyspace, self.table)}
+
+
+    def __build_schema(self):
+        """Really more of a config than a schema, but we'll allow it"""
+        root = {}
+
+        rampup_scenario = "run driver=cql tags==phase:rampup cycles===TEMPLATE(rampup-cycles,1000) threads=auto"
+        main_scenario = "run driver=cql tags==phase:main cycles===TEMPLATE(main-cycles,1000) threads=auto"
+        root["scenarios"] = {"default":[rampup_scenario, main_scenario]}
+        
+        root["bindings"] = build_bindings(self.table)
+
+        cl_map = {"cl":"LOCAL_QUORUM"}
+        cl_ratio_map = {"ratio":5}
+        cl_ratio_map.update(cl_map)
+
+        rampup_block = {"name":"rampup", "tags":{"phase":"rampup"}, "params":cl_map, "statements": [self.__build_rampup_statement()]}
+        main_read_block = {"name":"main-read", "tags":{"phase":"main", "type":"read"}, "params":cl_ratio_map, "statements": [self.__build_main_read_statement()]}
+        main_write_block = {"name":"main-write", "tags":{"phase":"main", "type":"write"}, "params":cl_ratio_map, "statements": [self.__build_main_write_statement()]}
```

Comment:
and here:

```
- name: main-write
  params: *id001
  statements:
```

Reply:
This is a reference to the anchor we declared earlier.  The earlier Wikipedia citation has some additional info here as well.

All of this is legit YAML (it shows up in the official spec) and nosqlbench appears to run fine with this config so the YAML parser in play there seems okay with it.  I can do a bit of checking to see if this feature can be disabled for PyYAML but I don't think it's essential to do so.